tests/test_default_values.py

```diff
@@ -0,0 +1,214 @@
+        ):
+            self.assertEqual(expected_name, config_owner.value.name)
+            self.assertEqual(expected_location, config_owner.value.location)
+
+        # Named dict within list
+        incidents_data = (("whereabouts", 0), ("somewhere else", 1))
+        self.assertEqual(len(incidents_data), len(config_suite.snapshot.incidents))
+        for (expected_location, expected_casualties), config_owner in zip(
+            incidents_data, sorted(config_suite.snapshot.incidents)
+        ):
+            self.assertEqual(expected_location, config_owner.location)
+            self.assertEqual(expected_casualties, config_owner.casualties)
+
+    def test_element_transformations_applied(self):
+        raw_config = car.build_all_default_config()
+        car_schema = car.build_schema()
+        tire_schema = car_schema[MK.Content]["tire"]
+        tire_schema[MK.Content]["dimension"][MK.Transformation] = car.inch_to_cm
+        config_suite = configsuite.ConfigSuite(raw_config, car_schema)
+
+        self.assertTrue(config_suite.valid)
+        self.assertAlmostEqual( 17 * 2.54, config_suite.snapshot.tire.dimension)
+
+    def test_context_transformation_applied(self):
+        raw_config = car.build_all_default_config()
+        car_schema = car.build_schema()
+        tire_schema = car_schema[MK.Content]["tire"]
+        tire_schema[MK.Content]["dimension"][MK.ContextTransformation] = car.inch_to_cm
+        config_suite = configsuite.ConfigSuite(raw_config, car_schema)
+
+        self.assertTrue(config_suite.valid)
+        self.assertAlmostEqual( 17 * 2.54, config_suite.snapshot.tire.dimension)
+
+    def test_element_validators_applied(self):
+        raw_config = car.build_all_default_config()
+        car_schema = car.build_schema()
+        tire_schema = car_schema[MK.Content]["tire"]
+        tire_schema[MK.Content]["dimension"][MK.ElementValidators] = [car.is_valid_dimension]
+
+        config_suite = configsuite.ConfigSuite(raw_config, car_schema)
+
+        self.assertFalse(config_suite.valid)
+        for err in config_suite.errors:
+            self.assertTrue(err.msg.startswith("Is x a valid dimension"))
+
+    def test_context_validators_applied(self):
+        raw_config = car.build_default_config()
+        car_schema = car.build_schema()
+
+        date_content = {MK.Type: types.DateTime,
+                        MK.Default: datetime.datetime(1999, 1, 1),
+                        MK.ContextValidators: [car.is_valid_date],
+                        MK.Required: False}
+
+        owner_schema = car_schema[MK.Content]["owner"]
+        owner_schema[MK.Content][MK.Value][MK.Content]["date"] = date_content
+
+        config_suite = configsuite.ConfigSuite(raw_config, car_schema, extract_validation_context=car.extract_production_date,)
+
+        self.assertFalse(config_suite.valid)
+        for err in config_suite.errors:
+            self.assertTrue(err.msg.startswith("Is x a valid date"))
```

Comment:
Can you also check the type of the errors whenever there are errors?

Reply:
I sure can